lib/machine/yaml/mapGoals.ts

```diff
@@ -81,83 +78,89 @@ const MapContainer: MapGoal = async (goals: any,
                                      goalMakers: Record<string, GoalMaker>,
                                      additionalTests: Record<string, PushTest>,
                                      extensionTests: Record<string, PushTestMaker>) => {
-    let goal: Goal;
     if (!!goals.containers) {
+
+        if (!goals.name) {
+            throw new Error(`Property 'name' missing on container goal:\n${JSON.stringify(goals, undefined, 2)}`);
+        }
+
         const containers = [];
-        const name = _.get(goals, "containers.name") || _.get(goals, "containers[0].name");
         for (const gc of goals.containers) {
             containers.push({
-                ...camelCase(gc),
+                ...gc,
                 name: gc.name.replace(/ /g, "-"),
                 test: !!gc.test ? await mapTests(gc.test, additionalTests, extensionTests) : undefined,
             });
         }
-        goal = container(
-            name,
+        const g = container(
+            goals.name,
             {
                 callback: containerCallback(),
                 containers,
-                volumes: camelCase(toArray(goals.volumes)),
-                input: toArray(goals.input),
-                output: mapOutput(goals),
```

Comment:
So the handling of input/output on the fulfillment is now all that is needed?

Reply:
The inputs and outputs are registered for any yaml goal now in addCaching. Now this is consistent with all goals coming from yaml.